components/tikv_util/src/logger/mod.rs

```diff
@@ -238,6 +260,57 @@ where
     }
 }
 
+pub struct RocksFormat<D>
+where
+    D: Decorator,
+{
+    decorator: D,
+}
+
+impl<D> RocksFormat<D>
+where
+    D: Decorator,
+{
+    pub fn new(decorator: D) -> Self {
+        Self { decorator }
+    }
+}
+
+impl<D> Drain for RocksFormat<D>
+where
+    D: Decorator,
+{
+    type Ok = ();
+    type Err = io::Error;
+
+    fn log(&self, record: &Record<'_>, values: &OwnedKVList) -> Result<Self::Ok, Self::Err> {
+        self.decorator.with_record(record, values, |decorator| {
+            if !record.tag().ends_with("_header") {
+                decorator.start_timestamp()?;
+                write!(
+                    decorator,
+                    "[{}][{}]",
+                    chrono::Local::now().format(TIMESTAMP_FORMAT),
+                    thread::current().id().as_u64(),
+                )?;
+                decorator.start_level()?;
+                write!(decorator, "[{}]", get_unified_log_level(record.level()))?;
+                decorator.start_whitespace()?;
+                write!(decorator, " ")?;
+            }
+            decorator.start_msg()?;
+            let msg = format!("{}", record.msg());
```

Comment:
I suspect if here would introduce one more memory copy, maybe check `ends_with` directly on `record.msg()`?

Reply:
`record.msg()` is of type `fmt::Arguments` which cannot used as str.